### PR TITLE
docs: fix some typos

### PR DESCRIPTION
**Pull request by danepitkin**
Improve the prose of documentation and improve some examples.

### CODE REVIEW

File: site/docs/expressions/scalar_functions.md [hunk 1]
```diff
@@ -134,7 +134,7 @@ For reference, here are are some common output type derivations and how they can
 
 | Operation                                                    | Definition                                                   |
 | ------------------------------------------------------------ | ------------------------------------------------------------ |
-| Add item to list                                             | `add(<List<T>, T>) => List<T>`                               |
+| Add item to list                                             | `add(List<T>, T) => List<T>`                               |
```
Comment (danepitkin) [inline site/docs/expressions/scalar_functions.md:137]: I assume this is the correct syntax, but let me know if I'm mistaken.

File: site/docs/expressions/specialized_record_expressions.md [hunk 2]
```diff
@@ -43,7 +43,7 @@ Switch expression allow a selection of alternate branches based on the value of
 ```
 switch(value)
 <value 1> => <return 1> (1 or more times)
-<else> => <return 3>
+<else> => <return n> (the nth return option)
```
Comment (danepitkin) [inline site/docs/expressions/specialized_record_expressions.md:46]: Do you think this communicates the idea more clearly than the original documentation?

File: site/docs/relations/basics.md [hunk 3]
```diff
@@ -19,7 +19,7 @@ In functions, function signatures are declared externally to the use of those si
 Each relational operation must declare the following:
 
 * Transformation logic around properties of the data. For example, does a relational operation maintain sortedness of a field? Does an operation change the distribution of data? 
-* How many input sets does an operation produce?
+* How many input sets does an operation require?
```
Comment (danepitkin) [inline site/docs/relations/basics.md:22]: Is this change correct?

**Review by danepitkin — COMMENTED** [review]
I'm new to substrait, so I highlighted a few areas where a second pair of eyes would help to ensure I'm correcting the wording properly.

File: site/docs/expressions/scalar_functions.md (hunk 1 above)
Comment (westonpace) [reply site/docs/expressions/scalar_functions.md:137]: Yes, you are correct here.

File: site/docs/expressions/extended_expression.md [hunk 4]
```diff
@@ -1,19 +1,19 @@
-# Extended expression
+# Extended Expression
 
-Extended expression is provided for expression level protocol instead of plan rels. It mainly targets for expression only evaluation, such as those computed in Filter/Project/Aggregation rels. Different from original expression defined in substrait protocol, it requires more information to completely describe the computation context, including input data schema, referred function signatures and output schema.
```
[the remark this answers is not in the document]
Comment (danepitkin) [reply site/docs/expressions/extended_expression.md:3]: Thanks for the clarification! This makes sense. I had assumed `Plan` was the top level root node for `ExtendedExpressions` also.

File: site/docs/expressions/extended_expression.md [hunk 5]
```diff
@@ -1,19 +1,19 @@
-# Extended expression
+# Extended Expression
 
-Extended expression is provided for expression level protocol instead of plan rels. It mainly targets for expression only evaluation, such as those computed in Filter/Project/Aggregation rels. Different from original expression defined in substrait protocol, it requires more information to completely describe the computation context, including input data schema, referred function signatures and output schema. 
+Extended Expression messages are provided for expression-level protocols as an alternative to using a Plan. They mainly target expression-only evaluations, such as those computed in Filter/Project/Aggregation rels. Different from the original Expression defined in the substrait protocol, Extended Expression messages require more information to completely describe the computation context including: input data schema, referred function signatures, and output schema.
```
Comment (westonpace) [inline site/docs/expressions/extended_expression.md:3]:
```suggestion
Extended Expression messages are provided for expression-level protocols as an alternative to using a Plan. They mainly target expression-only evaluations, such as those computed in Filter/Project/Aggregation rels. Unlike the original Expression defined in the substrait protocol, Extended Expression messages require more information to completely describe the computation context including: input data schema, referred function signatures, and output schema. 
```

File: site/docs/expressions/specialized_record_expressions.md [hunk 6]
```diff
@@ -14,7 +14,7 @@ These expressions allow structs, lists, and maps to be constructed from a set of
 ## Cast Expression
 To convert a value from one type to another, Substrait defines a cast expression. Cast expressions declare an expected type, an input argument and an enumeration specifying failure behavior, indicating whether cast should return null on failure or throw an exception.
 
-Note that Substrait always requires a cast expression whenever the current type is not exactly equal to (one of) the expected types. For example, it is illegal to directly pass a value of type `i8[0]` to a function that only supports an `i8?[0]` argument, some other variation of `i8`, or `i16[0]`.
+Note that Substrait always requires a cast expression whenever the current type is not exactly equal to (one of) the expected types.
```
Comment (westonpace) [inline site/docs/expressions/specialized_record_expressions.md:17]: Well, I did kind of like the i8 / i8? example.  This point is kind of subtle and so I believe the example was very intentionally placed.  It was only the discussion of "variants" (which is a different concept from nullability) that I thought was incorrect.

**Review by westonpace — CHANGES_REQUESTED** [review]
Two more small things.

File: site/docs/expressions/specialized_record_expressions.md (hunk 6 above)
Comment (danepitkin) [reply site/docs/expressions/specialized_record_expressions.md]: Added back! Sorry, I had misunderstood your original comment.

**Review by westonpace — APPROVED** [review]
Thank you for helping clarify all of this!

**Comment by danepitkin** [comment]
My first PR! Thanks for all of the help :)

**Pull request merged**